lib/lbt/analyzer/ComponentAnalyzer.js

```diff
@@ -116,12 +117,47 @@ class ComponentAnalyzer {
 
 		// TODO usages
 
+		// See sap/ui/core/Component._createManifestModelConfigurations
 		each( ui5.models, (options, model) => {
+			let modelType;
 			if ( options.type ) {
-				const module = ModuleName.fromUI5LegacyName( options.type );
-				log.verbose("derived model implementation dependency ", module);
-				info.addDependency(module);
+				modelType = options.type;
+			} else if ( options.dataSource && sapApp.dataSources ) {
+				const oDataSource = sapApp.dataSources[options.dataSource];
+				if (!oDataSource) {
+					return;
```

Comment:
Maybe log a warning that a model is declared in the manifest which can't be matched?

Reply:
```suggestion
log.warning(`Provided dataSource ${options.dataSource} for model "${model}" is unknown.`);
return;
```